src/pages/spawn/spawnVolume/SpawnVolumeButton.tsx

```diff
@@ -1,33 +1,57 @@
 import { useState } from "react";
 import styled from "@emotion/styled";
 import { palette } from "@leafygreen-ui/palette";
+import Tooltip from "@leafygreen-ui/tooltip";
 import { Disclaimer } from "@leafygreen-ui/typography";
 import { useSpawnAnalytics } from "analytics/spawn/useSpawnAnalytics";
 import { PlusButton } from "components/Buttons";
-import { size } from "constants/tokens";
-import { useSpruceConfig } from "hooks";
+import { size, zIndex } from "constants/tokens";
 import { SpawnVolumeModal } from "./spawnVolumeButton/SpawnVolumeModal";
 
-export const SpawnVolumeButton: React.VFC = () => {
+interface SpawnVolumeButtonProps {
+  maxSpawnableLimit: number;
+  volumeLimit: number;
+}
+
+export const SpawnVolumeButton: React.VFC<SpawnVolumeButtonProps> = ({
+  maxSpawnableLimit,
+  volumeLimit,
+}) => {
   const [openModal, setOpenModal] = useState(false);
   const spawnAnalytics = useSpawnAnalytics();
-  const spruceConfig = useSpruceConfig();
-  const volumeLimit = spruceConfig?.providers?.aws?.maxVolumeSizePerUser;
+
   return (
     <PaddedContainer>
-      <PlusButton
-        data-cy="spawn-volume-btn"
-        onClick={() => {
-          setOpenModal(true);
-          spawnAnalytics.sendEvent({ name: "Opened the Spawn Volume Modal" });
-        }}
+      <Tooltip
+        align="top"
+        justify="middle"
+        triggerEvent="hover"
+        popoverZIndex={zIndex.tooltip}
+        trigger={
+          <div>
+            <PlusButton
+              data-cy="spawn-volume-btn"
+              disabled={maxSpawnableLimit === 0}
+              onClick={() => {
+                setOpenModal(true);
+                spawnAnalytics.sendEvent({
+                  name: "Opened the Spawn Volume Modal",
+                });
+              }}
+            >
+              Spawn a Volume
+            </PlusButton>
+          </div>
+        }
+        enabled={maxSpawnableLimit === 0}
       >
-        Spawn a Volume
-      </PlusButton>
+        {`You have reached the max volume limit (${volumeLimit} GiB). Delete some volumes to spawn more.`}
+      </Tooltip>
       <Info>Limit {volumeLimit} GiB per User</Info>
       <SpawnVolumeModal
```

Comment:
Only including the modal in the DOM when it's open can make it easier to reset the form and error states if you like. Something like:

```js
{openModal && <SpawnVolumeModal />
```

Reply:
this was a great suggestion 🥰 ty!